examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts

```diff
@@ -16,47 +16,40 @@ import {
   upsertObservation,
 } from './intake-utils';
 
-export async function handler(event: BotEvent<QuestionnaireResponse>, medplum: MedplumClient): Promise<void> {
+export async function handler(medplum: MedplumClient, event: BotEvent<QuestionnaireResponse>): Promise<void> {
   const response = event.input;
 
-  const questionnaire = await medplum.readResource('Questionnaire', (response.questionnaire as string).split('/')[1]);
   const answers = getQuestionnaireAnswers(response);
 
   if (!response.subject) {
-    return;
+    throw new Error('Missing subject');
   }
 
   const patient = await medplum.readReference(response.subject as Reference<Patient>);
 
   if (!patient) {
-    return;
+    throw new Error('Patient not found');
   }
 
   // Handle demographic information
 
-  const newName = {
-    given: [answers['first-name'].valueString, answers['middle-name'].valueString],
-    family: answers['last-name'].valueString,
-  } as HumanName;
-
-  patient.name = [newName];
-  patient.birthDate = answers['dob'].valueDate;
-  patient.gender = answers['gender-identity'].valueCoding?.code as Patient['gender'];
+  patient.name = [
+    {
+      given: [answers['first-name']?.valueString, answers['middle-name']?.valueString],
```

Comment:
```
 answers['middle-name']?.valueString],
```

This potentially adds `undefined` to the array. we should filter out this undefined

Reply:
Move this logic to a function to make name processing more fine grained